src/unix/linux_like/linux/mod.rs

```diff
@@ -805,6 +941,45 @@ s_no_extra_traits! {
         pub d_type: ::c_uchar,
         pub d_name: [::c_char; 256],
     }
+
+    #[cfg(libc_union)]
+    pub union iwreq_data {
+            pub name: [c_char; ::IFNAMSIZ],
+            pub essid: iw_point,
+            pub nwid: iw_param,
+            pub freq: iw_freq,
+            pub sens: iw_param,
+            pub bitrate: iw_param,
+            pub txpower: iw_param,
+            pub rts: iw_param,
+            pub frag: iw_param,
+            pub mode: __u32,
+            pub retry: iw_param,
+            pub encoding: iw_point,
+            pub power: iw_param,
+            pub qual: iw_quality,
+            pub ap_addr: ::sockaddr,
+            pub addr: ::sockaddr,
+            pub param: iw_param,
+            pub data: iw_point,
+    }
+
+    pub struct iw_event {
+        pub len: __u16,
+        pub cmd: __u16,
+        pub u: iwreq_data,
+    }
+
+    #[cfg(libc_union)]
+    pub union __c_anonymous_iwreq {
+        pub ifrn_name: [c_char; ::IFNAMSIZ],
+    }
+
+    pub struct iwreq {
+        #[cfg(libc_union)]
+        pub ifr_ifrn: __c_anonymous_iwreq,
```

Comment:
"similarly" if libc_union is not set then iwreq will have the wrong size due to the missing field.